### PR TITLE
node: Refactor config to accept Firehose provider 

### PR DESCRIPTION
**Pull request by maoueh**
(and made Provide a little bit more extensible along the way)

There is a backward compatible handling of config that at some point should be converted to the new syntax form. The new syntax form is

```toml
[chains]
[chains.ropsten]
provider = [
  { label = "firehose", details = { type = "web3", transport = "rpc", url = "https://ropsten.sf.io/rpc", features = [] } },
]
```

While a Firehose provider now looks like:

```toml
[chains]
[chains.ropsten]
provider = [
  { label = "firehose", details = { type = "firehose", url = "https://ropsten.streamingfast.io" } },
]
```

This is for now undocumented and is first step towards Firehose integration in the codebase directly.

### CODE REVIEW

**Comment by maoueh** [comment]
@otaviopace @tilacog When you have a few spare minutes, you like a review here. Feel free to ping someone else with more knowledge of this part if required.

**Comment by leoyvens** [comment]
I think @lutter is the ideal person to review this.

**Review by lutter — APPROVED** [review]
Very nice! I really like that it's backwards compatible, and that the tests are pretty extensive.

**Comment by maoueh** [comment]
Excellent! Can we have it merged now hehe :0

**Comment by otaviopace** [comment]
@maoueh the way we usually do is that once a PR is approved, we let for the author to merge it whenever is better for them  :slightly_smiling_face:

The cases we want/need to wait for something else to go first or depend on another thing (maybe running the integration tests on that branch), people usually warn in the approve comment.

I would just double check if it's fine with @leoyvens in this case because I remember him mentioning that for one of your PRs he was waiting on something related to the tokio one to review or merge (I'm not quite sure honestly).

**Comment by leoyvens** [comment]
@otaviopace the thing is @maoueh doesn't have merge permissions. But it's a good idea give #2679 priority just in case there are any conflicts.

**Comment by maoueh** [comment]
Totally independent from #2679, anyway, I'm pushing the merge conflicts into @jubeless hands :D

**Comment by maoueh** [comment]
@otaviopace It's good, we got access now, so I'm able to approve now.

**Pull request merged**
